lib/init.coffee

```diff
@@ -11,4 +15,36 @@ module.exports =
         'Available codes: https://pypi.python.org/pypi/pep257#error-codes')
 
   activate: ->
-    console.log 'activate linter-pep257'
+    @subscriptions = new CompositeDisposable
+    @subscriptions.add atom.config.observe 'linter-pep257.executablePath',
+      (executablePath) =>
+        @executablePath = executablePath
+    @subscriptions.add atom.config.observe 'linter-pep257.ignoreCodes',
+      (ignoreCodes) =>
+        @ignoreCodes = ignoreCodes
+
+  deactivate: ->
+    @subscriptions.dispose()
+
+  provideLinter: ->
+    provider =
+      grammarScopes: ['source.python']
+      scope: 'file' # or 'project'
+      lintOnFly: true # must be false for scope: 'project'
+      lint: (textEditor)->
```

Comment:
I am getting failures with `@executablePath` resolving to none.

The issue here here, notice this bit of code is within an object and so when `lint` is used as a callback, the scope is within the provider (not the exported module). When creating callbacks, if you need to reference class variables, one must bind the scope. 

The lint function needs some help. The code would be far more understandable if broken into smaller routines that have a [single responsibility](https://en.wikipedia.org/wiki/Single_responsibility_principle).

Maybe something like:

```js
  lintPath: ->
    ...

  parseMessages: (output) ->
    ...

  lint: (textEditor) ->
    @filePath = textEditor.getPath()
    outputPromise = @lintPath() 
    messages = outputPromise.then(@parseMessages.bind(@))
    return messages

  provideLinter: ->
    provider =
      grammarScopes: ['source.python']
      scope: 'file'
      lintOnFly: true
      lint: @lint.bind(@)
```